src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -912,6 +907,18 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
             return traversalInstance;
         }
 
+        private void EmitMetaProject(ProjectRootElement metaProject, string metaProjectPath)
+        {
+            string filePath = Environment.GetEnvironmentVariable("MSBuildEmitSolution") == null ? Path.GetTempFileName() : metaProjectPath;
+            metaProject.Save(filePath);
```

Comment:
Would it be reasonable to use the `ProjectRootElement.Save(TextWriter)` overload here to avoid the physical file write?